soh/soh/Enhancements/randomizer/location_access/dungeons/fire_temple.cpp

```diff
@@ -506,114 +631,174 @@ void RegionTable_Init_FireTemple() {
+        //Locations
+        LOCATION(RC_FIRE_TEMPLE_MQ_FIRE_PILLAR_LEFT_HEART,  true),
+        LOCATION(RC_FIRE_TEMPLE_MQ_FIRE_PILLAR_RIGHT_HEART, true),
+    }, {
+        //Exits
+        Entrance(RR_FIRE_TEMPLE_MQ_LAVA_GEYSER_1F,      []{return true;}),
+        Entrance(RR_FIRE_TEMPLE_MQ_LAVA_GEYSER_PILLARS, []{return true;}),
     });
 
-    areaTable[RR_FIRE_TEMPLE_MQ_BIG_TORCH_ROOM] = Region("Fire Temple MQ Big Torch Room", SCENE_FIRE_TEMPLE, {}, {}, {
+    //tunic logic handled on entry
+    areaTable[RR_FIRE_TEMPLE_MQ_LAVA_GEYSER_PILLARS] = Region("Fire Temple MQ Lava Geyser Pillars", SCENE_FIRE_TEMPLE, {}, {
+        //Locations
+        //rang can get the other 2 hearts from here but it's an awkward shot so a trick
+        LOCATION(RC_FIRE_TEMPLE_MQ_FIRE_PILLAR_LOWER_HEART, true),
+    }, {
         //Exits
-        Entrance(RR_FIRE_TEMPLE_MQ_LOWER_MAZE,         []{return (logic->HasFireSource() && logic->CanUse(RG_HOOKSHOT)) || (ctx->GetTrickOption(RT_FIRE_MQ_CLIMB) && logic->CanUse(RG_HOVER_BOOTS));}),
-        Entrance(RR_FIRE_TEMPLE_MQ_ELEVATOR_ROOM,      []{return logic->CanUse(RG_GORON_TUNIC);}),
-        Entrance(RR_FIRE_TEMPLE_MQ_MAZE_SHORTCUT_CAGE, []{return logic->Get(LOGIC_FIRE_OPENED_UPPER_SHORTCUT);}),
+        Entrance(RR_FIRE_TEMPLE_MQ_LAVA_GEYSER_1F,    []{return true;}),
+        Entrance(RR_FIRE_TEMPLE_MQ_LAVA_GEYSER_GRATE, []{return logic->CanUse(RG_HOOKSHOT);}),
+        Entrance(RR_FIRE_TEMPLE_MQ_LAVA_GEYSER_2F,    []{return true;}),
     });
 
-    areaTable[RR_FIRE_TEMPLE_MQ_LOWER_MAZE] = Region("Fire Temple MQ Lower Maze", SCENE_FIRE_TEMPLE, {}, {
-        //Locations
-        //Check handled on both floors
-        LOCATION(RC_FIRE_TEMPLE_MQ_LIZALFOS_MAZE_SIDE_ROOM_CHEST, logic->HasExplosives() && ctx->GetTrickOption(RT_FIRE_MQ_MAZE_SIDE_ROOM)),
-     }, {
+    areaTable[RR_FIRE_TEMPLE_MQ_LAVA_GEYSER_2F] = Region("Fire Temple MQ Lava Geyser 2F", SCENE_FIRE_TEMPLE, {}, {
+    }, {
+        //Exits
+        //the block lift does some janky stuff that makes leave sideways while on it difficult, so you can't avoid fall damage if you want to land on pillars.
+        //you can still move "backwards" into the door though and with a roll jump onto the grate, which avoids damage
+        Entrance(RR_FIRE_TEMPLE_MQ_LAVA_GEYSER_1F,      []{return logic->FireTimer() >= 24;}),
+        Entrance(RR_FIRE_TEMPLE_MQ_LAVA_GEYSER_PILLARS, []{return logic->TakeDamage() && logic->FireTimer() >= 24;}),
+        Entrance(RR_FIRE_TEMPLE_MQ_SHORTCUT_ROOM,       []{return true;}),
+    });
+
+    areaTable[RR_FIRE_TEMPLE_MQ_SHORTCUT_ROOM] = Region("Fire Temple MQ Shortcut Room", SCENE_FIRE_TEMPLE, {}, {}, {
+        //Exits
+        Entrance(RR_FIRE_TEMPLE_MQ_LOWER_LIZALFOS_MAZE, []{return (logic->HasFireSource() && logic->CanUse(RG_HOOKSHOT) /*&& (logic->IsAdult || CanUse(RG_CLIMB))*/) || (ctx->GetTrickOption(RT_FIRE_MQ_CLIMB) && logic->CanUse(RG_HOVER_BOOTS)/*&& CanUse(RG_CLIMB)*/);}),
```

Comment:
```suggestion
        Entrance(RR_FIRE_TEMPLE_MQ_LOWER_LIZALFOS_MAZE, []{return (logic->HasFireSource() && (logic->IsAdult || logic->CanUse(RG_HOOKSHOT)) /*&& (logic->IsAdult || CanUse(RG_CLIMB))*/) || (ctx->GetTrickOption(RT_FIRE_MQ_CLIMB) && logic->CanUse(RG_HOVER_BOOTS)/*&& CanUse(RG_CLIMB)*/);}),
```
adult can jump across & climb hookshot pillar

(rest needs some adjustment since there's a hookshot target on ceiling to skip climb)

Reply:
split the room into 3 regions to both help keep that clean and for decoupled doorsanity.